core/src/main/java/com/linecorp/armeria/server/http/auth/OAuth1aTokenExtractor.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.http.auth;
+
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+
+import com.linecorp.armeria.common.http.HttpHeaderNames;
+import com.linecorp.armeria.common.http.HttpHeaders;
+
+/**
+ * Extracts {@link OAuth1aToken} from {@link HttpHeaders}, in order to be used by
+ * {@link HttpAuthServiceBuilder}.
+ */
+public class OAuth1aTokenExtractor implements Function<HttpHeaders, OAuth1aToken> {
+
+    private static final Logger logger = LoggerFactory.getLogger(OAuth1aTokenExtractor.class);
+    private static final Pattern AUTHORIZATION_HEADER_PATTERN = Pattern.compile(
+            "\\s*(?i)oauth\\s+(?<parameters>\\S+)\\s*");
+
+    @Override
+    public OAuth1aToken apply(HttpHeaders headers) {
+        String authorization = headers.get(HttpHeaderNames.AUTHORIZATION);
+        if (Strings.isNullOrEmpty(authorization)) {
+            logger.warn("Authorization header does not exist");
```

Comment:
Fixed, along with BasicTokenExtractor & OAuth2TokenExtractor (which you omitted to comment).